edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Core/Agent.cs

```diff
@@ -131,7 +132,9 @@ public async Task ReconcileAsync(CancellationToken token)
                     else
                     {
                         ModuleSet desiredModuleSet = deploymentConfig.GetModuleSet();
-                        this.availabilityMetric.ComputeAvailability(desiredModuleSet, current);
+                        _ = Task.Run(() => this.availabilityMetric.ComputeAvailability(desiredModuleSet, current))
```

Comment:
Do we need to have some way of cancelling this when we shutdown?

Reply:
It is a void that completes quickly, so it should;d be fine